ui/apps/platform/src/Containers/Network/Header/useNamespaceFilters.ts

```diff
@@ -13,55 +12,19 @@ const selector = createStructuredSelector<SelectorState, SelectorResult>({
     selectedNamespaceFilters: selectors.getSelectedNamespaceFilters,
 });
 
-type NamespaceMetadataResp = {
-    id: string;
-    results: {
-        namespaces: {
-            metadata: {
-                name: string;
-            };
-        }[];
-    };
-};
-
-export const NAMESPACES_FOR_CLUSTER_QUERY = gql`
-    query getClusterNamespaceNames($id: ID!) {
-        results: cluster(id: $id) {
-            id
-            namespaces {
-                metadata {
-                    name
-                }
-            }
-        }
-    }
-`;
-
 function useNamespaceFilters() {
-    const [availableNamespaceFilters, setAvailableNamespaceFilters] = useState<string[]>([]);
     const { selectedClusterId, selectedNamespaceFilters } = useSelector<
         SelectorState,
         SelectorResult
     >(selector);
-    // If the selectedClusterId has not been set yet, do not run the gql query
-    const queryOptions = selectedClusterId
-        ? { variables: { id: selectedClusterId } }
-        : { skip: true };
 
-    const { loading, error, data } = useQuery<NamespaceMetadataResp, { id: string }>(
-        NAMESPACES_FOR_CLUSTER_QUERY,
-        queryOptions
+    const targetClusterId: string | undefined =
+        selectedClusterId == null ? undefined : selectedClusterId;
```

Comment:
what is the goal of transforming `null` to `undefined`?

Reply:
The goal here is to match the signature of the `useFetchClusterNamespacesForPermissions` function.
I changed the function signature to accept string, null or undefined and removed that code piece.